scala/dllib/src/main/scala/com/intel/analytics/bigdl/dllib/example/nnframes/xgboost/xgbClassifierTrainingExample.scala

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2016 The BigDL Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.intel.analytics.bigdl.dllib.examples.nnframes.xgboost
+
+import ml.dmlc.xgboost4j.scala.spark.TrackerConf
+
+import com.intel.analytics.bigdl.dllib.NNContext
+import com.intel.analytics.bigdl.dllib.nnframes.XGBClassifier
+
+import org.apache.spark.ml.feature.{StringIndexer, VectorAssembler}
+import org.apache.spark.sql.{SQLContext, SparkSession}
+import org.apache.spark.sql.types.{DoubleType, StringType, StructField, StructType}
+object xgbClassifierTrainingExample {
+  def main(args: Array[String]): Unit = {
+    if (args.length < 4) {
+      println("Usage: program input_path num_threads num_round modelsave_path")
+      sys.exit(1)
+    }
+    val sc = NNContext.initNNContext()
+    val spark = SQLContext.getOrCreate(sc)
+
+    val inputPath = args(0)
+    val num_threads = args(1).toInt
+    val num_round = args(2).toInt
+    val modelsave_path= args(3)
+
+    val schema = new StructType(Array(
+      StructField("sepal length", DoubleType, true),
+      StructField("sepal width", DoubleType, true),
+      StructField("petal length", DoubleType, true),
+      StructField("petal width", DoubleType, true),
+      StructField("class", StringType, true)))
+    val rawInput = spark.read.schema(schema).csv(inputPath)
```

Comment:
We can handle preprocessing after read CSV. Spark dataframe is quite similar to pandas dataframew

Reply:
It seems that we can use iris.data directly. No need to preprocess.